packages/react/src/form/error-summary/focus.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Focus a form field while showing the associated legend or label in the viewport.
+ *
+ * Gives the user a better context of what field they are focusing on.
+ *
+ * Copied from https://github.com/alphagov/govuk-frontend/blob/cbf4ef1e329711be5b78a92bda6ba84a7db9ca40/packages/govuk-frontend/src/govuk/components/error-summary/error-summary.mjs#L60-L108
+ */
+export function focusWithLegendOrLabelInViewport(id: string) {
+  const input = document.getElementById(id);
+  if (!input) {
+    return false;
+  }
+
+  const legendOrLabel = maybeLegendForInput(input) ?? labelForInput(input);
+  if (!legendOrLabel) {
+    return false;
+  }
+
+  legendOrLabel.scrollIntoView();
+  input.focus({ preventScroll: true });
+
+  return true;
+}
+
+function maybeLegendForInput(input: HTMLElement) {
```

Comment:
It can techincally be other things than `input`, e.g. `select`. But that doesn't matter much for the purpose of this code as we just need to do `.focus` on the element, which all `HTMLElement` supports. Which is what `document.getElementById` returns. So this was some lazyness to keep typescript happy.